lib/resourcemerge/machineconfig.go

```diff
@@ -78,6 +79,37 @@ func ensureControllerConfigSpec(modified *bool, existing *mcfgv1.ControllerConfi
 	setBytesIfSet(modified, &existing.KubeAPIServerServingCAData, required.KubeAPIServerServingCAData)
 	setBytesIfSet(modified, &existing.CloudProviderCAData, required.CloudProviderCAData)
 
+	// First off: do I need to generally account for Infra changes? Seems like we should?
+	if required.Infra != nil && !equality.Semantic.DeepEqual(existing.Infra, required.Infra) {
+		*modified = true
+		existing.Infra = required.Infra
+	}
+
+	// Moving on... if all Infras are empty (ie born in 4.1 clusters), maybe update Infra with values we _do_ have(?)
+	if required.Infra == nil && existing.Infra == nil {
+		// from types.go: "platform is deprecated, use Infra.Status.PlatformStatus.Type instead"
+		if existing.Platform != "" {
+			*modified = true
+			existing.Infra = &configv1.Infrastructure{
+				Status: configv1.InfrastructureStatus{
+					PlatformStatus: &configv1.PlatformStatus{
+						// but errr do I need to make sure this platform is the correct constant (as seen in types_infrastructure.go)???
+						Type: configv1.PlatformType(existing.Platform),
+					},
+				},
+			}
+		}
+		// but alsooo from types.go: "etcdDiscoveryDomain is deprecated, use Infra.Status.EtcdDiscoveryDomain instead"
+		if existing.EtcdDiscoveryDomain != "" {
+			*modified = true
+			required.Infra = &configv1.Infrastructure{
+				Status: configv1.InfrastructureStatus{
+					EtcdDiscoveryDomain: existing.EtcdDiscoveryDomain,
+				},
+			}
+		}
+	}
+
```

Comment:
we can get rid of the "PlatformStatus.Type" check just below this also - we'll just sync the whole infra object as that's the source of truth anyway

Reply:
done